### PR TITLE
Small cleanups in async processor

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: sdk_contrib/async_processor/src/main/java/io/opentelemetry/sdk/contrib/trace/export/DisruptorEventQueue.java [hunk 1]
```diff
@@ -102,73 +102,82 @@ public DisruptorEvent newInstance() {
...
-    if (blocking) {
-      ringBuffer.publishEvent(TRANSLATOR_THREE_ARG, eventType, readableSpan, flushLatch);
-    } else {
-      // TODO: Record metrics if element not added.
-      ringBuffer.tryPublishEvent(TRANSLATOR_THREE_ARG, eventType, readableSpan, flushLatch);
+  void enqueueEndEvent(ReadableSpan span) {
+    if (isShutdown) {
+      if (!loggedShutdownMessage.getAndSet(true)) {
+        logger.info("Attempted to enqueue entry after Disruptor shutdown.");
+      }
```
Comment (jkwatson) [inline sdk_contrib/async_processor/src/main/java/io/opentelemetry/sdk/contrib/trace/export/DisruptorEventQueue.java]: this is duplicated code. can it be moved to a method?

File: sdk_contrib/async_processor/src/main/java/io/opentelemetry/sdk/contrib/trace/export/DisruptorEventQueue.java [hunk 2]
```diff
@@ -102,73 +102,82 @@ public DisruptorEvent newInstance() {
...
     }
+    enqueue(EventType.ON_END, span, null);
   }
 
-  // Shuts down the underlying disruptor.
+  // Shuts down the underlying disruptor. Ensures that when this method returns the disruptor is
+  // shutdown.
   void shutdown() {
-    enqueueAndLock(EventType.ON_SHUTDOWN);
+    synchronized (this) {
```
Comment (jkwatson) [inline sdk_contrib/async_processor/src/main/java/io/opentelemetry/sdk/contrib/trace/export/DisruptorEventQueue.java]: synchronizing on `this` is potentially dangerous, as you don't have control over who else might try to grab your monitor and keep this from ever executing.

**Comment by carlosalberto** [comment]
Overall LGTM, once @jkwatson's issues have been solved.

File: sdk_contrib/async_processor/src/main/java/io/opentelemetry/sdk/contrib/trace/export/DisruptorEventQueue.java (hunk 2 above)
Comment (bogdandrutu) [reply sdk_contrib/async_processor/src/main/java/io/opentelemetry/sdk/contrib/trace/export/DisruptorEventQueue.java]: Customized the messages to be more relevant.

File: sdk_contrib/async_processor/src/main/java/io/opentelemetry/sdk/contrib/trace/export/DisruptorEventQueue.java (hunk 1 above)
Comment (bogdandrutu) [reply sdk_contrib/async_processor/src/main/java/io/opentelemetry/sdk/contrib/trace/export/DisruptorEventQueue.java]: Customized the messages to be more relevant.

File: sdk_contrib/async_processor/src/main/java/io/opentelemetry/sdk/contrib/trace/export/DisruptorEventQueue.java (hunk 2 above)
Comment (bogdandrutu) [reply sdk_contrib/async_processor/src/main/java/io/opentelemetry/sdk/contrib/trace/export/DisruptorEventQueue.java]: This is an internal class so I would not worry that much about that.

Comment (bogdandrutu) [reply sdk_contrib/async_processor/src/main/java/io/opentelemetry/sdk/contrib/trace/export/DisruptorEventQueue.java]: @jkwatson are you ok with my comment?

Comment (jkwatson) [reply sdk_contrib/async_processor/src/main/java/io/opentelemetry/sdk/contrib/trace/export/DisruptorEventQueue.java]: yeah, I'm not convinced that it's a pattern we want to perpetuate, but it's ok for now.

**Pull request merged**